internal/checker/checker.go

```diff
@@ -498,15 +487,21 @@ func filterTrees(ctx context.Context, backlog restic.IDs, loaderChan chan<- rest
 		job                     treeJob
 		nextTreeID              restic.ID
 		outstandingLoadTreeJobs = 0
+		processedTrees          = make(map[restic.ID]bool)
```

Comment:
The `processedTrees` map will increase the memory usage of the check command as it is stored in addition to `blobrefs.M`. It would be possible to add `processedTrees` to `c.blobrefs` and use it exclusively to store the tree ids.  The main drawback of this would be that `UnusedBlobs` has to look into both maps.